### PR TITLE
ROX-26275: cluster reassignment admin API

### PR DESCRIPTION
**Pull request by johannes94**
## Description

This PR is the first of several PRs to enable migrating a tenant from on dataplane cluster to the other. It adds:
* A feature flag `RHACS_CLUSTER_MIGRATION`
* A admin API endpoint to be able to reassign the `cluster_id` column of a tenant in the DB via API

## Checklist (Definition of Done)

- [x] Unit and integration tests added
- [x] Added test description under `Test manual`
- [x] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [x] CI and all relevant tests are passing
- [x] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- [ ] Add secret to app-interface Vault or Secrets Manager if necessary
- [ ] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)
- [ ] Check AWS limits are reasonable for changes provisioning new resources
- [ ] (If applicable) Changes to the dp-terraform Helm values have been reflected in the addon on integration environment

## Test manual

* Added a new integration test
* CI should be sufficient for now, because the new endpoint is hidden behind a feature flag
* More e2e testing will be done once other parts required for actual cluster reassignment will be added

### CODE REVIEW

File: internal/dinosaur/test/integration/admin_test.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+	helper, adminClient, teardown := test.NewAdminHelperWithHooks(t, ocmServer, nil)
+	defer teardown()
+
+	orgID := "13640203"
+
+	centrals := []*dbapi.CentralRequest{
+		{
+			MultiAZ:        clusters[0].MultiAZ,
+			Owner:          "assigclusteruser1",
```
Comment (ebensh) [inline internal/dinosaur/test/integration/admin_test.go:39]: nit - assig?

File: internal/dinosaur/test/integration/admin_test.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
+package integration
```
Comment (ebensh) [inline internal/dinosaur/test/integration/admin_test.go:1]: Is it worth adding a test for trying to move between clusters that do not match criteria (e.g. different MultiAZ)? I didn't see one in the cluster_placement_strategy and wasn't sure if it would go there or here

**Review by ebensh — CHANGES_REQUESTED** [review]
LGTM, just some small nits; thanks for the cleanup along the way (renaming Dinosaur to Central) and making the review straightforward.

How did you feel about the amount of work to make a new API endpoint and change the cluster ID? It seems like a large change across many files, but was it fairly straightforward where to change what?

File: internal/dinosaur/pkg/services/cluster_placement_strategy.go [hunk 3]
```diff
@@ -50,6 +47,23 @@ func (d FirstReadyPlacementStrategy) FindCluster(central *dbapi.CentralRequest)
 	return nil, nil
 }
 
+// AllMatchingClustersForCentral returns all cluster that fit the criteria to run a central
+func (d FirstReadyPlacementStrategy) AllMatchingClustersForCentral(central *dbapi.CentralRequest) ([]*api.Cluster, error) {
```
[the remark this answers is not in the document]
Comment (johannes94) [reply internal/dinosaur/pkg/services/cluster_placement_strategy.go:51]: You're right sounds like that is more like a function that might be shared across multiple strategies. Moved it to a package function instead.

File: internal/dinosaur/test/integration/admin_test.go (hunk 1 above)
Comment (johannes94) [reply internal/dinosaur/test/integration/admin_test.go:39]: Whoops. Fixed. :)

File: internal/dinosaur/test/integration/admin_test.go (hunk 2 above)
Comment (johannes94) [reply internal/dinosaur/test/integration/admin_test.go:1]: Hard to tell where to put them, I'd say if I add one it's gonna be here because it tests the entire API to DB flow.

I have a test with a region mismatch, essentially the logic is the same. Do you think I should add more edge cases to that test?

**Comment by johannes94** [comment]
> LGTM, just some small nits; thanks for the cleanup along the way (renaming Dinosaur to Central) and making the review straightforward.
> 
> How did you feel about the amount of work to make a new API endpoint and change the cluster ID? It seems like a large change across many files, but was it fairly straightforward where to change what?

It looks worse than it is. Essentially you're modifying the openapi doc, then most things get generate by `make generate`.

The actual manual changes are fairly simple:
- Add a HTTP handler.
- Route to that handler.
- Add a function to the dinosaurservice that orchestrates business logic.
- Done.

**Comment by johannes94** [comment]
/retest

Cluster creation failed.

**Review by ebensh — APPROVED** [review]
